cylc/flow/task_job_mgr.py

```diff
@@ -239,48 +239,58 @@ def submit_task_jobs(self, suite, itasks, curve_auth,
-        auth_itasks = {}  # {install target: [itask, ...], ...}
+        auth_itasks = {}  # {platform: [itask, ...], ...}
         for itask in prepared_tasks:
-            install_target = get_install_target_from_platform(itask.platform)
-            auth_itasks.setdefault(install_target, [])
-            auth_itasks[install_target].append(itask)
+            platform_name = itask.platform['name']
+            auth_itasks.setdefault(platform_name, [])
+            auth_itasks[platform_name].append(itask)
         # Submit task jobs for each platform
         done_tasks = bad_tasks
-        for install_target, itasks in sorted(auth_itasks.items()):
-            ri_map = self.task_remote_mgr.remote_init_map
-            # Re-fetch a copy of platform
+
+        for platform_name, itasks in sorted(auth_itasks.items()):
             platform = itasks[0].platform
-            # Skip both remote init and remote file install for localhost
-            if (install_target == 'localhost' or
-                    not is_remote_host(get_host_from_platform(platform))):
-                LOG.debug(f"REMOTE INIT NOT REQUIRED for {install_target}")
-                ri_map[install_target] = (REMOTE_FILE_INSTALL_DONE)
-            # Remote init not in progress, start it
-            if install_target not in ri_map.keys():
-                self.task_remote_mgr.remote_init(
-                    platform, curve_auth, client_pub_key_dir)
-                for itask in itasks:
-                    itask.set_summary_message(self.REMOTE_INIT_MSG)
-                    self.job_pool.add_job_msg(
-                        get_task_job_id(
-                            itask.point, itask.tdef.name, itask.submit_num),
-                        self.REMOTE_INIT_MSG)
-                continue
-            # Already done remote so move on to file install
-            elif (ri_map[install_target] == REMOTE_INIT_DONE):
-                self.task_remote_mgr.file_install(platform)
-            # Already doing remote init or file install
-            elif (ri_map[install_target] in self.IN_PROGRESS.keys()):
-                for itask in itasks:
-                    msg = self.IN_PROGRESS[ri_map[install_target]]
-                    itask.set_summary_message(msg)
-                    self.job_pool.add_job_msg(
-                        get_task_job_id(
-                            itask.point, itask.tdef.name, itask.submit_num),
-                        msg)
-                continue
+            install_target = get_install_target_from_platform(platform)
+            ri_map = self.task_remote_mgr.remote_init_map
+
+            # Skip to jobs (use .get() to avoid KeyError)
+            if (ri_map.get(install_target) != REMOTE_FILE_INSTALL_DONE):
+                # Skip both remote init and remote file install for localhost
+                if (install_target == 'localhost' or
+                        not is_remote_host(get_host_from_platform(platform))):
```

Comment:
If another platform (on the same filesystem) comes along, it will not do the remote init as it will get caught by this condition.... 
https://github.com/cylc/cylc-flow/blob/41544aef16dfece2db10e4fe3f5fb7750625c3c2/cylc/flow/task_job_mgr.py#L284 

If one platform fails (tasks are submit-failed) but if another platform (same fs) comes along with the same install target I was told that that platform needs a chance of remote init.

Reply:
Unless this is the mis-configuration you mean... where you have:
```
platform frozen-1
host = anna
install target = olaf

platform frozen-2
host = anna
install target = sven 
```
This would attempt the remote init once for each platform but it will fail on the second as there will already be a mis-matched key on `anna`.

Reply:
So I think there is no need to check `is_remote_host(get_host_from_platform(platform))`?